modules/pcd/src/lib/parse-pcd.ts

```diff
@@ -10,15 +10,55 @@
 
 import {Schema, Field, Float32, Uint8, FixedSizeList} from '@loaders.gl/schema';
 import {getMeshBoundingBox} from '@loaders.gl/loader-utils';
-
-const LITTLE_ENDIAN = true;
-
-export default function parsePCD(data) {
+import {SchemaMetadata} from '@loaders.gl/schema/lib/schema/classes/schema';
+
+type IncomingHeader = {
+  data: any;
+  headerLen: number;
+  str: string;
+  version: RegExpExecArray | null | number;
+  fields: RegExpExecArray | null | string[];
+  size: RegExpExecArray | null | number[];
+  type: RegExpExecArray | null | string[];
+  count: RegExpExecArray | null | number[];
+  width: RegExpExecArray | number;
+  height: RegExpExecArray | number;
+  viewpoint: RegExpExecArray | null | string;
+  points: RegExpExecArray | number;
+  offset: {[index: string]: number};
+  rowSize: number;
+  boundingBox?: [];
+};
+type Normalized = {
+  POSITION: {
+    value: Float32Array;
+    size: number;
+  };
+  NORMAL?: {
+    value: Float32Array;
+    size: number;
+  };
+  COLOR_0?: {
+    value: Uint8Array;
+    size: number;
+  };
+};
+type HeaderAttributes = {
+  [index: string]: number[];
+};
+const LITTLE_ENDIAN: boolean = true;
+
+/**
+ *
+ * @param data
+ * @returns
+ */
+export default function parsePCD(data: ArrayBufferLike) {
   // parse header (always ascii format)
   const textData = new TextDecoder().decode(data);
   const pcdHeader = parsePCDHeader(textData);
 
-  let attributes;
+  let attributes: any = {};
```

Comment:
Can we avoid any? At least do `{[attributeName: string]: any}`?

Reply:
> {[attributeName: string]: any}

In this function we will convert **attributes**  below with the function **'getNormalizedAttributes'**
so I changed **HeaderAttributes** at the top of the file like you suggested and here is like an announcement of the variable